src/Build/BuildCheck/API/BuildAnalyzerConfiguration.cs

```diff
@@ -42,4 +46,55 @@ public class BuildAnalyzerConfiguration
     /// If some rules are enabled and some are not, the analyzer will be run and reports will be post-filtered.
     /// </summary>
     public bool? IsEnabled { get; internal init; }
+
+    /// <summary>
+    /// Creates a <see cref="BuildAnalyzerConfiguration"/> object based on the provided configuration dictionary.
+    /// If key, equals to the name of the property in lowercase, exists in the dictionary => the value is parsed and assigned to the instance property value.
+    /// </summary>
+    /// <param name="configDictionary">The configuration dictionary containing the settings for the build analyzer.</param>
+    /// <returns>A new instance of <see cref="BuildAnalyzerConfiguration"/> with the specified settings.</returns>
+    public static BuildAnalyzerConfiguration Create(Dictionary<string, string> configDictionary)
+    {
+        return new()
+        {
+            EvaluationAnalysisScope = TryExtractValue(nameof(EvaluationAnalysisScope).ToLower(), configDictionary, out EvaluationAnalysisScope evaluationAnalysisScope) ? evaluationAnalysisScope : null,
+            Severity = TryExtractValue(nameof(Severity).ToLower(), configDictionary, out BuildAnalyzerResultSeverity severity) ? severity : null,
+            IsEnabled = TryExtractValue(nameof(IsEnabled).ToLower(), configDictionary, out bool isEnabled) ? isEnabled : null,
+        };
+    }
+
+    private static bool TryExtractValue<T>(string key, Dictionary<string, string> config, out T value) where T : struct
+    {
+        value = default;
+
+        if (config == null || !config.ContainsKey(key))
+        {
+            return false;
+        }
+
+        bool isParsed = false;
+
+        if (typeof(T) == typeof(bool))
```

Comment:
nit: I think this would better be done by overload resolution at run-time. The logic in this method is simple and run-time type checks like this one are in this case unnecessary. Instead of a generic method with a `struct` constraint, please consider having two overloads. One taking `out bool value`, another taking `out T value` with the `struct, System.Enum` constraint.

Reply:
This is updated and switched to the overloading, thank you for the suggestion!